Sources/SwiftDocC/Infrastructure/Symbol Graph/SymbolGraphLoader.swift

```diff
@@ -176,34 +206,80 @@ struct SymbolGraphLoader {
 
         return (symbolGraph, isMainSymbolGraph)
     }
+    
+    /// Adds the missing fallback and default availability information to the unified symbol graph
+    /// in case it didn't exists in the loaded symbol graphs.
+    private func addMissingAvailability(
+        unifiedGraph: inout UnifiedSymbolGraph,
+        unconditionallyUnavailablePlatformNames: [PlatformName],
+        registeredPlatforms: [PlatformName],
+        defaultAvailabilities: [DefaultAvailability.ModuleAvailability]
+    ) {
+        // The fallback platforms that are missing from the unified graph correspond to
+        // the fallback platforms that have not been registered yet,
+        // are not marked as unavailable,
+        // and the corresponding inheritance platform has a SGF (has been registered).
+        let missingFallbackPlatforms = DefaultAvailability.fallbackPlatforms.filter {
+            !registeredPlatforms.contains($0.key) &&
+            !unconditionallyUnavailablePlatformNames.contains($0.key) &&
+            registeredPlatforms.contains($0.value)
+        }
+        // Platforms that are defined in the Info.plist that had no corresponding SGF
+        // and are not being added as fallback of another platform.
+        let missingAvailabilities = defaultAvailabilities.filter {
+                !missingFallbackPlatforms.keys.contains($0.platformName) &&
+                !registeredPlatforms.contains($0.platformName)
+        }
+        let symbolsWithFallbackAvailability = unifiedGraph.symbols.compactMapValues { symbol  -> UnifiedSymbolGraph.Symbol in
+            for (selector, _) in symbol.mixins {
+                if var symbolAvailability = (symbol.mixins[selector]?["availability"] as? SymbolGraph.Symbol.Availability) {
+                    guard !symbolAvailability.availability.isEmpty else { continue }
+                    // Add fallback availability.
+                    for (fallbackPlatform, inheritedPlatform) in missingFallbackPlatforms {
+                        if !symbolAvailability.availability.contains(where: { $0.domain?.rawValue == fallbackPlatform.rawValue }) {
+                            symbolAvailability.availability.forEach {
+                                // Add the platform fallback to the availability mixin the platform is inheriting from.
+                                // The added availability only copies the introduced version.
+                                if $0.domain?.rawValue.lowercased() == inheritedPlatform.rawValue.lowercased() {
```

Comment:
There are a few different places that make this check, some with a case sensitive comparison and some with a case <em><b>in</b></em>sensitive comparison. 

Maybe it would be worth adding a couple of private extensions in this file to make all these places use the same case sensitivity? For example:
```swift
private extension SymbolGraph.Symbol.Availability {
    func contains(_ platform: PlatformName) -> Bool {
        availability.contains(where: { $0.matches(platform) })
    }
}

private extension SymbolGraph.Symbol.Availability.AvailabilityItem {
    func matches(_ platform: PlatformName) -> Bool {
        domain?.rawValue.lowercased() == platform.rawValue.lowercased()
    }
}
```

Reply:
Implemented here https://github.com/apple/swift-docc/commit/c55abc329895424275b96a3c47afef3f8c5bb3f9